### PR TITLE
LyX: remove Qt variants and fix Qt4

### PR DESCRIPTION
**Pull request by kurthindenburg**
Per ML, variants can't handle different versions.  So remove variants and just use os version to pick Qt version.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: aqua/LyX/Portfile [hunk 1]
```diff
@@ -6,9 +6,62 @@ PortGroup           boost 1.0
...
+                        sha256  0d5a75dec9a169ab23bfa47da53ec65c62432de57014de315023d620ebccfe68 \
+                        size    17871168
+
+    qt5.depends_component       qtbase \
+                                qtmacextras \
+                                qtsvg
+
+    set branch          [join [lrange [split ${version} .] 0 1] .]
+
+    configure.cxxflags-append -std=c++11
```
Comment (reneeotten) [inline aqua/LyX/Portfile:52]: this implies you need a compiler that supports the C++11 standard, but further down you request C++14 compatible compiler; that is inconsistent.

File: aqua/LyX/Portfile [hunk 2]
```diff
@@ -6,9 +6,62 @@ PortGroup           boost 1.0
...
+    PortGroup qt5 1.0
+
+    version             2.4.2
+    revision            0
+
+    checksums           rmd160  686630c5fe6d98bb29625e0818d9ad027bdb3814 \
+                        sha256  0d5a75dec9a169ab23bfa47da53ec65c62432de57014de315023d620ebccfe68 \
+                        size    17871168
+
+    qt5.depends_component       qtbase \
```
Comment (reneeotten) [inline aqua/LyX/Portfile:46]: qtbase is implied and should not be listed

**Review by reneeotten — CHANGES_REQUESTED** [review]
Besides the inline comments two other ones:
- remove the "python38" variant as that is an EOL Python version; and "boost" doesn't have these variants anymore. 
- whitespace is very inconsistent and does not always appear to follow the modeline

File: aqua/LyX/Portfile [hunk 3]
```diff
@@ -94,60 +136,6 @@ variant python312 conflicts python38 python39 python310 python311 description {U
...
-                        --with-qt-includes=${prefix}/libexec/qt5/include \
-                        --with-qt-libraries=${prefix}/libexec/qt5/lib \
-                        --enable-qt5
-}
-
-# error: '-fobjc-exceptions' is required to enable Objective-C exception syntax
-if {[string match *gcc* ${configure.compiler}]} {
-    # This must be set in pre-configure:
-    pre-configure {
-        configure.objcflags-append \
```
Comment (barracuda156) [inline aqua/LyX/Portfile:146]: This should not be removed.

**Review by barracuda156 — CHANGES_REQUESTED** [review]

File: aqua/LyX/Portfile [hunk 4]
```diff
@@ -6,9 +6,60 @@ PortGroup           boost 1.0
...
+                    size    16412700
+
+    set branch      [join [lrange [split ${version} .] 0 1] .]
+
+    configure.pre_args  --prefix=${applications_dir}/LyX.app
+    configure.args-append \
+                    --without-x \
+                    --without-included-boost \
+                    --enable-optimization=-Os \
+                    --with-version-suffix=-${branch} \
```
Comment (ryandesign) [inline aqua/LyX/Portfile:28]: There's some code duplication. These lines seem to be the same regardless of version. Can this common code be pulled out of the conditional?

An additional question is why `-Os` is being hardcoded. Why not use `${configure.optflags}`?

**Review by ryandesign — COMMENTED** [review]
The build fails in CI on macOS 13 and 14:

```
  insets/ExternalTransforms.cpp:334:20: error: 'any_cast<std::function<std::unique_ptr<const lyx::external::TransformOption> (std::string)>>' is unavailable: introduced in macOS 10.13
    334 |         Factory factory = any_cast<Factory>(any_factory);
        |                           ^
```

I think that's because it is using `-mmacosx-version-min=10.10`; we don't want to request such an old deployment target. Oddly, the macOS 15 build succeeds despite this setting.

File: aqua/LyX/Portfile [hunk 5]
```diff
@@ -37,117 +88,56 @@ master_sites        http://lyx.cybermirror.org/stable/${branch}.x/ \
...
+if {![variant_isset python39] &&
     ![variant_isset python310] && ![variant_isset python311]} {
     default_variants-append +python312
 }
 
 # TODO: use loops like other ports do
-variant python38 conflicts python39 python310 python311 python312 description {Use python38} {
-    depends_build-append    port:python38
-    configure.python        ${prefix}/bin/python3.8
-}
```
Comment (ryandesign) [inline aqua/LyX/Portfile:75]: If adding a python313 variant is possible, that would be a good idea.

**Comment by kurthindenburg** [comment]
Thanks for the comments - I'll fix those once I get it to build correctly

**Comment by kurthindenburg** [comment]
The 15 CI is struggling to build this huge port.  Unless someone objects, I'm going to commit it.  I'll squash the commit message.

**Pull request merged**